### ISSUE TITLE
test_zh_TW_format fails in 2.10.2 sdist (outdated locale data?)

### ISSUE THREAD

**Issue by mgorny** [python-babel/babel#886]
## Overview Description

When running the Babel test suite inside 2.10.2 extracted from the pypi sdist, I get the following test failure:

```
FAILED tests/test_dates.py::test_zh_TW_format - AssertionError: assert 'B12:34:56' == '中午12:34:56'
```

## Steps to Reproduce

1. `wget https://files.pythonhosted.org/packages/b0/91/c521fa4102ef9afc8c9f64d1828993d39b7e99944d40a76d4ef94f6321b2/Babel-2.10.2.tar.gz`
2. `tar -xf Babel-2.10.2.tar.gz`
3. `cd Babel-2.10.2`
4. `pytest tests/test_dates.py::test_zh_TW_format`

## Actual Results

```pytb
$ pytest tests/test_dates.py::test_zh_TW_format 
========================================================= test session starts =========================================================
platform linux -- Python 3.9.13, pytest-7.1.2, pluggy-1.0.0
rootdir: /tmp/Babel-2.10.2, configfile: setup.cfg
plugins: cov-3.0.0
collected 1 item                                                                                                                      

tests/test_dates.py F                                                                                                           [100%]

============================================================== FAILURES ===============================================================
__________________________________________________________ test_zh_TW_format __________________________________________________________

    def test_zh_TW_format():
        # Refs GitHub issue #378
>       assert dates.format_time(datetime(2016, 4, 8, 12, 34, 56), locale='zh_TW') == u'中午12:34:56'
E       AssertionError: assert 'B12:34:56' == '中午12:34:56'
E         - 中午12:34:56
E         ? ^^
E         + B12:34:56
E         ? ^

tests/test_dates.py:836: AssertionError
======================================================= short test summary info =======================================================
FAILED tests/test_dates.py::test_zh_TW_format - AssertionError: assert 'B12:34:56' == '中午12:34:56'
========================================================== 1 failed in 0.53s ==========================================================
```

## Expected Results

Tests passing ;-).

## Reproducibility

Stable failure.

## Additional Information

Regenerating the CLDR data (via `make`) seems to fix the problem. I suppose this means that the sdist archive ships outdated data.

**Comment by akx** [comment]
Fixed in Babel 2.10.3. Sorry again.

**Comment by mgorny** [comment]
No problem. Thanks a lot!
